cmd/checkertools/new.go

```diff
@@ -0,0 +1,126 @@
+package checkertools
+
+import (
+	"bytes"
+	"flag"
+	"go/build"
+	"go/format"
+	"html/template"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+// TODO: add more such tools, like checker-rename to easily rename
+// checker type, update comments and rename testdata dir?
+
+// NewMain implements gocritic sub-command entry point.
+func NewMain() {
+	flag.Usage = func() {
+		log.Printf("usage: rulename")
+		flag.PrintDefaults()
+	}
+	flag.Parse()
+
+	if flag.NArg() != 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+	// TODO: add camel case check.
```

Comment:
I think we can add note in usage about camel case style and if name doesn't fit proper style we can cover this during review before merging 😉